api/lib/infrastructure/repositories/target-profile-for-update-repository.js

```diff
@@ -1,23 +1,43 @@
 import { knex } from '../../../db/knex-database-connection.js';
+import { DomainTransaction } from '../DomainTransaction.js';
 
-const update = async function ({
-  targetProfileId,
-  name,
-  imageUrl,
-  description,
-  comment,
-  category,
-  areKnowledgeElementsResettable,
-}) {
-  const targetProfileToUpdate = {
+const TARGET_PROFILE_TABLE_NAME = 'target-profiles';
+const TARGET_PROFILE_TUBES_TABLE_NAME = 'target-profile_tubes';
+
+const update = async function (targetProfile, domainTransaction = DomainTransaction.emptyTransaction()) {
+  const knexConn = domainTransaction?.knexTransaction || knex;
```

Comment:
Ici la transaction ne me semble plus facultative car il y a plusieurs instructions de modifications. J'enlèverai bien l'instruction `DomainTransaction.emptyTransaction()` pour rendre cela obligatoire. Ou alors remplacer cette ligne par

```suggestion
  // Créée une transaction si aucune n'a été fourni
  const knexConn = domainTransaction?.knexTransaction || (await knex.transaction());
```

Reply:
Pour enlever le `DomainTransaction.emptyTransaction()` il faut d'abord vérifier que cette méthode est pas utilisé dans un autre usecase que celui concerné par la PR 🤔

Reply:
Je me basé sur [ce qui existe pour le `training-trigger-repository`](https://github.com/1024pix/pix/blob/dev/api/src/devcomp/infrastructure/repositories/training-trigger-repository.js#L17), ça a l'air de fonctionner sans casse !